ethereum/stratum/src/test/java/org/hyperledger/besu/ethereum/stratum/StratumConnectionTest.java

```diff
@@ -87,20 +89,21 @@ public void testStratum1Matches() {
                 + "  \"params\": [ "
                 + "    \"MinerName/1.0.0\", \"EthereumStratum/1.0.0\" "
                 + "  ]"
-                + "}\n"));
+                + "}\n"),
+        message::set);
     assertThat(called.get()).isFalse();
 
     assertThat(message.get())
         .isEqualTo(
-            "{\"jsonrpc\":\"2.0\",\"id\":23,\"result\":[[\"mining.notify\",\"abcd\",\"EthereumStratum/1.0.0\"],\"\"]}\n");
+            "{\"jsonrpc\":\"2.0\",\"id\":23,\"result\":[[\"mining.notify\",\"abcd\",\"EthereumStratum/1.0.0\"],\"\"]}");
```

Comment:
each message must end with a \n character.

Reply:
I moved the messages split one layer up by adding [`io.netty.handler.codec.LineBasedFrameDecoder`](https://netty.io/4.0/api/io/netty/handler/codec/LineBasedFrameDecoder.html) to the channel pipeline. That's why now the `StratumConnection` only serves as a bridge between the message and the _protocols_ and doesn't need to handle line endings anymore.

Reply:
The decoder decodes, but how about sending messages?

Reply:
You were correct, the original PR used `System.lineSeparator()` as the delimiter for responses which wasn't suitable for defining a protocol as this was OS dependant. I changed it to always be delimited by `\n` as it is in main. You can see this in `StratumServer#sendLineBasedResponse`.
Also, there is a test covering this type of responses in `StratumServerTest#handlePlainTextMultiple`.